homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -119,6 +129,24 @@
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the Xiaomi Miio components from a config entry."""
     hass.data.setdefault(DOMAIN, {})
+    host = entry.data[CONF_HOST]
+
+    # Create push server
+    if KEY_PUSH_SERVER not in hass.data[DOMAIN]:
+        push_server = PushServer(host)
```

Comment:
Will the server work for all kinds of devices, using the gateway and not using the gateway?

Reply:
Yes in principle this is a generic miio protocol and this schould work for all xiaomi miio devices.
So gateways but also vacuum robots, fans, humidifiers, sockets etc.

However for now it is only implemented for the gateway.
I am receiving a vacuum robot this week so will probably add push events to the vacuums once this is merged.

Reply:
We're setting up the server for all devices currently, not only the gateway.

Reply:
I know, that is for future implementation for the other devices.
For other devices the server will just sit there and do nothing since it will not receive any messages. Since it is async I don't think it will have any perfomance impact if it is set up and does not receive any messages.

But then other devices such as the vacuums I am planning to implement already have the server setup.

Reply:
I already have a roborock vacuum and will be implementing the push server also for the vacuums once this is merged, therefore sticking to the general setup of the server.

Reply:
We shouldn't create a server that's not used.

Reply:
I now only setup the server for gateway devices.